tests/openwisp2/settings.py

```diff
@@ -96,6 +97,8 @@
 STATIC_URL = '/static/'
 MEDIA_URL = '/media/'
 MEDIA_ROOT = '{0}/media/'.format(BASE_DIR)
+PRIVATE_STORAGE_ROOT = '/{0}/firmware/'.format(BASE_DIR)
+PRIVATE_STORAGE_CLASS = 'openwisp_firmware_upgrader.private_storage.storage.OpenWISPPrivateFileSystemStorage'
```

Comment:
why is this setting here? Is it needed for django-private-storage?

Reply:
I'd set those two settings as defaults in https://github.com/openwisp/openwisp-firmware-upgrader/blob/master/openwisp_firmware_upgrader/apps.py#L12, allowing them to be overridden.
I'll do it later if you don't get to do it now.